jenkins/scripts/files/run_integration_tests.sh

```diff
@@ -53,47 +53,58 @@ fi
 
 if [ "${REPO_NAME}" == "metal3-dev-env" ]
 then
-   export METAL3REPO="${UPDATED_REPO}"
-   export METAL3BRANCH="${UPDATED_BRANCH}"
-   export BAREMETAL_OPERATOR_LOCAL_IMAGE="https://github.com/metal3-io/baremetal-operator.git"
-   export CAPM3_LOCAL_IMAGE="https://github.com/metal3-io/cluster-api-provider-metal3.git"
-   if [ "${CAPI_VERSION}" == "v1alpha3" ]
-   then
-     export CAPM3_LOCAL_IMAGE_BRANCH="release-0.3"
-   else
-     export CAPM3_LOCAL_IMAGE_BRANCH="master"
-   fi
+  export METAL3REPO="${UPDATED_REPO}"
+  export METAL3BRANCH="${UPDATED_BRANCH}"
+
+  # If the target repo and branch are the same as the source repo and branch
+  # we're running a master test, that is not for a PR, so we build the image
+  # for CAPM3 to verify the process (not BMO due to the build time for BMO image)
+  if [[ "${UPDATED_BRANCH}" == "${REPO_BRANCH}" ]] && [[ "${UPDATED_REPO}" != *"${REPO_ORG}/${REPO_NAME}"* ]]; then
```

Comment:
Why UPDATED_REPO should be the same as REPO_BRANCH? Doesn't it mean that it is a master job?
Maybe it should be like `"${UPDATED_BRANCH}" != "${REPO_BRANCH}" `?

Reply:
if UPDATED_BRANCH is equal to REPO_BRANCH, and UPDATED_REPO is similar to REPO_ORG/REPO_NAME, it means that it is not a PR so it is a master job.